merlin/task_servers/task_server_factory.py

```diff
@@ -0,0 +1,258 @@
+            LOG.info(f"Created {server_type} task server")
+            return instance
+        except Exception as e:
+            raise MerlinInvalidTaskServerError(
+                f"Failed to create {server_type} task server: {e}"
+            ) from e
+
+    def register(self, name: str, server_class: TaskServerInterface, 
+                aliases: List[str] = None) -> None:
+        """
+        Register a new task server implementation.
+
+        Args:
+            name: The canonical name for the task server.
+            server_class: The class implementing TaskServerInterface.
+            aliases: Optional list of alternative names for this task server.
+
+        Raises:
+            TypeError: If the server_class does not implement TaskServerInterface.
+        """
+        if not issubclass(server_class, TaskServerInterface):
+            raise TypeError(f"{server_class} must implement TaskServerInterface")
+
+        self._task_servers[name] = server_class
+        LOG.debug(f"Registered task server: {name}")
+
+        if aliases:
+            for alias in aliases:
+                self._task_server_aliases[alias] = name
+                LOG.debug(f"Registered alias '{alias}' for task server '{name}'")
+
+    def get_server_info(self, server_type: str) -> Dict:
+        """
+        Get information about a specific task server type.
+
+        Args:
+            server_type: The name of the task server to get info for.
+
+        Returns:
+            Dictionary containing server information and capabilities.
+
+        Raises:
+            MerlinInvalidTaskServerError: If the requested task server is not supported.
+        """
+        # Resolve alias to canonical task server name
+        server_type = self._task_server_aliases.get(server_type, server_type)
+
+        if server_type not in self._task_servers:
+            self._discover_plugins()
+
+        if server_type not in self._task_servers:
+            available = ", ".join(self.list_available())
+            raise MerlinInvalidTaskServerError(
+                f"Task server '{server_type}' is not supported by Merlin. "
+                f"Available task servers: {available}"
+            )
+
+        server_class = self._task_servers[server_type]
+        return {
+            "name": server_type,
+            "class": server_class.__name__,
+            "module": server_class.__module__,
+            "description": server_class.__doc__ or "No description available",
+        }
+
+    def _discover_plugins(self) -> None:
+        """
+        Discover and load task server plugins from entry points and modules.
+        
+        This method attempts to find additional task server implementations
+        through Python entry points and module scanning.
+        """
+        # METHOD 1: Entry points (for pip-installable plugins)
+        try:
+            import pkg_resources
+            for entry_point in pkg_resources.iter_entry_points('merlin.task_servers'):
```

Comment:
This works up to python 3.11 (I think) but for 3.12+ we need to use importlib_metadata instead since they deprecated pkg_resources.